File: scala/6kyu - C.scala
// Link -> https://www.codewars.com/kata/54dc6f5a224c26032800005c

object Hello extends App{

  def stockSummary(lstOfArt: Array[String], lstOfCat: Array[String]): String = {
    var stock: Map[String, Int] = Map[String, Int]()
    var stockReport = ""

    if (lstOfArt.length == 0) return ""

    for (book <- lstOfArt) {
      val category: String = book.take(1)
      val quantity: Int = book.split(" ")(1).toInt

      if (stock.contains(category)) {
        val newValue = stock(category) + quantity
        stock = stock + (category -> newValue)
      }
      else {
        stock += (category -> quantity)
      }
    }

    for(category <- lstOfCat){
      if (!stock.contains(category)){
        stockReport = stockReport + " - " + "(" + category + " : 0" + ")"
      }
      else {
        stockReport = stockReport + " - " + "(" + category + " : " + stock(category) + ")"
      }
    }

    stockReport = stockReport.drop(2)

    return stockReport
  }
}

File: python/6kyu - C.py
# Link -> https://www.codewars.com/kata/541c8630095125aba6000c00

def digital_root(n):
    a = str(n)
    b = 0
    
    while len(a) > 1:
        for ch in a:
            b += int(ch)
        a = str(b)
        b = 0
    
    a = int(a)
    return a
